lib/uksignal/signal.c

```diff
@@ -297,20 +339,44 @@ int sigwait(const sigset_t *set, int *sig)
 	ptr->wait.status = UK_SIG_NOT_WAITING;
 
 	/* do not execute handler, set received signal */
-	*sig = ptr->wait.received_signal.si_signo;
+	*info = ptr->wait.received_signal;
 
 	/* execute other pending signals */
 	uk_sig_handle_signals();
 
 	return 0; /* returns positive errno */
 }
 
+#if UK_LIBC_SYSCALLS
+int sigwait(const sigset_t *set, int *sig)
+{
+	int error;
+
+	error = rt_sigtimedwait(set, &si, NULL, (_NSIG / 8));
+	*sig = si.si_signo;
+
+	return error;
+}
+#endif
+
+UK_SYSCALL_R_DEFINE(int, tkill, pid_t __unused, pid, int, sig)
```

Comment:
This issues a big preproccessing error on my end. For whatever reason, it does not like the name `tkill`. If I were to place some random name here, say `gkill`, that error does not occur anymore. Is this just on my end?

Reply:
Apparently the tkill symbol was not added to UK_PROVIDED_SYSCALLS, sorry for this, I think something was messed up during the rebase.